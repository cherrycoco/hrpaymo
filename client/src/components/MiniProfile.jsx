import React from 'react';
import Avatar from 'material-ui/Avatar';
import Divider from 'material-ui/Divider';
import Paper from 'material-ui/Paper';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';


const style = {
  card: {
    position: 'relative',
    width: '100%',
    display: 'inline-block',
    fontWeight: 700,
    fontSize: '18px',
  },
  title: {
    fontWeight: 700,
    fontSize: '20px',
    margin: '10px'
  },
  balance: {
    marginLeft: '125px',
    fontSize: '18px',
  }
};

class MiniProfile extends React.Component {
  constructor (props) {
    super(props);
  }

  render() {
    return (
      <div>
      <Paper className='feed-container' id='mini-profile'>
        <Card>
          <CardHeader
            title={
              <div>
                <span style={style.title}>{this.props.userInfo.displayName}</span>
              </div>
            }
            subtitle={
              <div className='member-tag'>
                <p>@{this.props.userInfo.username}</p>
                <p>{this.props.userInfo.email}</p>
              </div>
            }
            avatar={
              <Avatar 
                size={100} 
                src={this.props.userInfo.avatarUrl || '/images/no-image.gif'}
              />
            }
            />
        </Card>
      </Paper>
      </div>
    );
  }
}

export default MiniProfile;

