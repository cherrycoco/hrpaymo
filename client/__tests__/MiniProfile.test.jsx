import React from 'react';
import ReactDOM from 'react-dom';
import MiniProfile from '../src/components/MiniProfile.jsx';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

const userInfo = {
  avatarUrl: "",
  createdAt: "Thu Jan 18 2018 12:04:05 GMT-0800 (PST)",
  displayName: "Santi Iraola",
  userId: 31,
  username: "blam92",
  verified: true
}
it('Miniprofile renders without crashing (needs to be wrapped in MuiThemeProvider)', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MuiThemeProvider>
    <MiniProfile 
      balance={"100.00"}
      userInfo={userInfo} />
  </MuiThemeProvider>, div);
});

it('renders a snapshot of miniprofile', () => {
  const tree = renderer.create(
  <MuiThemeProvider>
    <MiniProfile 
      balance={"100.00"}
      userInfo={userInfo} />
  </MuiThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
});
let miniProfileComp = (<MuiThemeProvider>
    <MiniProfile 
      balance={"100.00"}
      userInfo={userInfo} />
  </MuiThemeProvider>);

it('should mount in a full DOM', function() {
  // expect(shallow(miniProfileComp)
  // .contains(<span></span>)).toBe(true);
  console.log(shallow(miniProfileComp).state());
});
