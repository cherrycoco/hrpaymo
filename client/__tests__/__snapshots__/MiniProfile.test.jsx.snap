// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of miniprofile 1`] = `
<div>
  <div
    className="feed-container"
    id="mini-profile"
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "fontWeight": 500,
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "whiteSpace": "nowrap",
            }
          }
        >
          <img
            className={undefined}
            size={100}
            src="/images/no-image.gif"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 50,
                "height": 100,
                "justifyContent": "center",
                "marginRight": 16,
                "muiPrepared": true,
                "userSelect": "none",
                "width": 100,
              }
            }
          />
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              <div>
                <span
                  style={
                    Object {
                      "fontSize": "20px",
                      "fontWeight": 700,
                      "margin": "10px",
                    }
                  }
                >
                  Santi Iraola
                </span>
              </div>
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            >
              <div
                className="member-tag"
              >
                <p>
                  @
                  blam92
                </p>
                <p />
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
